### ISSUE TITLE
DICompileUnit not listed in llvm.dbg.cu

### ISSUE THREAD

**Issue by glandium** [rust-lang/rust#90045]
STR:
- Create a dummy crate: `cargo new hello ; cd hello`
- Build with profiling and LTO enabled: `CARGO_INCREMENTAL=0 cargo +nightly rustc -- -Zprofile -Cdebuginfo=2 -Clto -Cembed-bitcode=yes`

This fails with the following error message:
```
DICompileUnit not listed in llvm.dbg.cu
!34 = distinct !DICompileUnit(language: DW_LANG_Rust, file: !13, producer: "clang LLVM (rustc version 1.58.0-nightly (bd41e09da 2021-10-18))", isOptimized: false, runtimeVersion: 0, emissionKind: FullDebug, enums: !35)
LLVM ERROR: Broken module found, compilation aborted!
```

Bisect says it is a regression from #89041.

**Comment by Nemo157** [comment]
This seems to compile successfully now on `rustc 1.65.0-nightly (29e4a9ee0 2022-08-10)`.

**Comment by Nilstrieb** [comment]
@glandium can you confirm that this works now?

**Comment by glandium** [comment]
Last broken: rustc 1.64.0-nightly (23e21bdd2 2022-07-15)
First working: rustc 1.64.0-nightly (d5e7f4782 2022-07-16)
(and yes, still works today)
Presumably, this was fixed by 98190b716829fd4edbf1b364a1a662de9a9ca7bb, which is a revert of the commit pointed at by bisect.
